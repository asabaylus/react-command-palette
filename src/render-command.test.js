import * as React from "react";
import Enzyme, { mount } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import serializer from "enzyme-to-json/serializer";

import allCommands from "./__mocks__/commands";
import RenderCommand from "./render-command";
import SampleCustomCommand from "../examples/sampleCustomCommand";
import DefaultCommand from "./default-command";

// React 16 Enzyme adapter
Enzyme.configure({ adapter: new Adapter() });

expect.addSnapshotSerializer(serializer);

describe("RenderCommand", () => {
  it("should render the default child command", () => {
    const wrapper = mount(<RenderCommand suggestion={allCommands[0]} />);
    expect(wrapper).toMatchSnapshot();
  });
  it("should render a custom child command", () => {
    const wrapper = mount(
      <RenderCommand
        suggestion={allCommands[0]}
        renderCommand={SampleCustomCommand}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
