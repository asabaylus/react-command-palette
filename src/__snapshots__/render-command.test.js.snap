// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RenderCommand should render a custom child command 1`] = `
<RenderCommand
  renderCommand={[Function]}
  suggestion={
    Object {
      "category": "Command",
      "command": [Function],
      "id": 1,
      "name": "Start All Data Imports",
    }
  }
>
  <div>
    <div
      className="item"
    >
      <span
        className="category Command"
      >
        Command
      </span>
      <span>
        Start All Data Imports
      </span>
    </div>
  </div>
</RenderCommand>
`;

exports[`RenderCommand should render the default child command 1`] = `
<RenderCommand
  suggestion={
    Object {
      "category": "Command",
      "command": [Function],
      "id": 1,
      "name": "Start All Data Imports",
    }
  }
>
  <DefaultCommand
    suggestion={
      Object {
        "category": "Command",
        "command": [Function],
        "id": 1,
        "name": "Start All Data Imports",
      }
    }
  >
    <div
      className="item"
    >
      <span>
        Start All Data Imports
      </span>
    </div>
  </DefaultCommand>
</RenderCommand>
`;
